### PR TITLE
Adding formulate testing and npm publish actions

### PR DESCRIPTION
**Pull request by bjmccotter7192**
Adding formulate testing action:
- Building the application using the setup-node action
    1. Runs ```yarn install``` to install all needed packages
    2. Runs ```yarn build``` to build and place all files in the appropriate place
    3. Runs ```yarn test``` to run all tests that will be used for branch requirement

Adding NPM publish action:
- Builds the application and pushes it to the NPM repo @department-of-veterans-affairs/formulate
    1. Runs ```yarn install```, ```yarn build```, then ```npm publish``` using NPM Auth Token

P.S. - The NPM Publish command will use the name specified in the package.json file to push to the correct package when we create a "Release" on this repo

### CODE REVIEW

**Review by micahchiang — APPROVED** [review]
I don't have much familiarity with github actions, but this all looks fairly self explanatory. When you have a sec, could you add a brief summary to the description for this PR so we have some historical context for it if we ever want to look back in the future? Thanks!

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        cache: 'yarn'
+    
+    - name: Install dependencies in react-components
+      run: yarn install
+    
+    - name: Build package NPM package
+      run: yarn build
+    
+    - name: Publish build artifacts to NPM
+      run: npm publish --access public --tolerate-republish
```
Comment (bkjohnson) [inline .github/workflows/npm-publish.yml:26]: It doesn't look like `npm` has `--tolerate-publish` as an option: https://docs.npmjs.com/cli/v6/commands/npm-publish

If you want to keep it, you would need to use [`yarn npm publish`](https://yarnpkg.com/cli/npm/publish)

**Review by bkjohnson — COMMENTED** [review]
Sorry for the delayed review :upside_down_face: 

Has this been tested? I didn't see a run where a publish was attempted.

File: .github/workflows/npm-publish.yml (hunk 1 above)
Comment (bjmccotter7192) [reply .github/workflows/npm-publish.yml:26]: Good find, I will research that and see if we need the tolerate-republish or if we can just use the suggested

**Pull request merged**